src/to_string.c

```diff
@@ -101,7 +93,47 @@ size_t number_to_string(char *const dest, uint64_t number) {
     return length;
 }
 
-#define DECIMAL_DIGITS 8
+#define DECIMAL_DIGITS 9
+#define NANO_AVAX_SCALE 1000000000
+
+// Display avax in human readable form
+size_t nano_avax_to_string(char *const dest, uint64_t nano_avax) {
+    check_null(dest);
+    uint64_t whole_avax = nano_avax / NANO_AVAX_SCALE;
+    uint64_t fractional_avax = nano_avax % NANO_AVAX_SCALE;
+    size_t off = number_to_string(dest, whole_avax);
+    if (fractional_avax == 0) {
+        return off;
+    }
+    dest[off++] = '.';
+
+    char tmp[MAX_INT_DIGITS];
+    convert_number(tmp, fractional_avax, true);
+
+    // Eliminate trailing 0s
+    char *start = tmp + MAX_INT_DIGITS - DECIMAL_DIGITS;
+    char *end;
+    for (end = tmp + MAX_INT_DIGITS - 1; end >= start; end--) {
+        if (*end != '0') {
+            end++;
+            break;
+        }
+    }
+
+    size_t length = end - start;
+    memcpy(dest + off, start, length);
+    off += length;
+    dest[off] = '\0';
+    return off;
+}
+
+void nano_avax_to_string_indirect64(char *const dest, size_t const buff_size, uint64_t const *const number) {
+    check_null(dest);
+    check_null(number);
+    if (buff_size < MAX_INT_DIGITS + 1)
```

Comment:
I think this needs to be `MAX_INT_DIGITS + 2`...  one for the null, one for a decimal point.   Otherwise we risk undefined behavior.  (I.e. this appears to be a "type 2" function when it should be a "type 1" function,  in the parlance of [John Regehr](https://blog.regehr.org/archives/213))

Also wondering if this routine should succeed or fail if the buffer isn't big enough for the largest possible int... but that's mostly idle speculation,  as I think this is fine.

Reply:
MAX_INT_DIGITS is 20, so we can cover up to 10^11 avax (with 10^-9 decimal). So yeah we can't cover anything amounts greater than that.

Reply:
Right,  a 64 bit integer can be 20 digits.   Then we have one null character at the end,  and one decimal point.   So the destination buffer needs to be be at least 22 bytes long.